### PR TITLE
Adding statistics to BWS

### PR DESCRIPTION
**Pull request by matiaspando**
Stats are available using 
http://localhost:3232/bws/api/v1/stats/:from/:to

For example
http://localhost:3232/bws/api/v1/stats/2015-04-21/2015-05-05

### CODE REVIEW

File: lib/expressapp.js [hunk 1]
```diff
@@ -314,11 +315,22 @@ ExpressApp.start = function(opts, cb) {
     });
   });
 
+  router.get('/v1/stats/:from/:to/', function(req, res) {
+    var opts = {};
+    opts.from = req.params['from'];
+    opts.to = req.params['to'];
+    var data = stats.getStats(opts, function(err, data) {
```
Comment (isocolsky) [inline lib/expressapp.js]: this `var data` is confusing...

File: scripts/stats.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    });
+  });
+};
+
+
+bwsStats.ProccesWallets = function(DB, cb) {
+  var collection = DB.collection('wallets');
+  collection.find({}).toArray(function(err, items) {
+    if (err) {
+      console.log('Error.', err);
```
Comment (isocolsky) [inline scripts/stats.js]: It should not continue processing on error

File: scripts/stats.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  if (opts) {
+    startDate = moment(opts.from);
+    endDate = moment(opts.to);
+  }
+  bwsStats.cleanUp();
+
+  mongodb.MongoClient.connect(url, function(err, db) {
+    if (err) {
+      console.log('Unable to connect to the mongoDB server. Error:', err);
+      return;
```
Comment (isocolsky) [inline scripts/stats.js]: There should be a call to the `cb` function passed as arg

File: scripts/stats.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    stats[data.network][day] = {
+      totalTx: 0,
+      totalAmount: 0,
+      totalNewWallets: 0
+    };
+  }
+  stats[data.network][day].totalNewWallets++;
+};
+
+bwsStats.TotalTxpForToday = function(data) {
```
Comment (isocolsky) [inline scripts/stats.js]: The name of this function is misleading. It computes txp stats for any day in the interval and not just for the current day.

**Comment by matiaspando** [comment]
Updated! Thanks for the comments @isocolsky

**Comment by matiaspando** [comment]
Updated!

**Comment by matiaspando** [comment]
Updated!

**Pull request merged**